src/common/data_type.rs

```diff
@@ -309,6 +313,7 @@ impl DataTypeMap {
             ScalarValue::DurationMillisecond(_) => Ok(DataType::Duration(TimeUnit::Millisecond)),
             ScalarValue::DurationMicrosecond(_) => Ok(DataType::Duration(TimeUnit::Microsecond)),
             ScalarValue::DurationNanosecond(_) => Ok(DataType::Duration(TimeUnit::Nanosecond)),
+            _ => todo!(),
```

Comment:
is there an alternative to panicking here?

Reply:
Yes, change it to return Err